tests/unit/s2n_resume_test.c

```diff
@@ -67,6 +67,225 @@ int main(int argc, char **argv)
+            EXPECT_SUCCESS(s2n_stuffer_free(&actual_data));
+        }
+
+        /* TLS1.3 with early data: all variable fields set */
+        {
+            struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+            conn->actual_protocol_version = S2N_TLS13;
+
+            /* Set non-zero length secret */
+            uint8_t secret_size = 0;
+            EXPECT_SUCCESS(s2n_hmac_digest_size(conn->secure.cipher_suite->prf_alg, &secret_size));
+            EXPECT_SUCCESS(s2n_alloc(&conn->tls13_ticket_fields.session_secret, secret_size));
+
+            /* Set early data fields */
+            const uint8_t data[] = "test data";
+            EXPECT_SUCCESS(s2n_connection_set_server_max_early_data_size(conn, 1));
+            EXPECT_MEMCPY_SUCCESS(conn->application_protocol, data, sizeof(data));
+            EXPECT_SUCCESS(s2n_connection_set_server_early_data_context(conn, data, sizeof(data)));
+
+            /* Result matches constants */
+            size_t actual_size = 0;
+            EXPECT_OK(s2n_connection_get_session_state_size(conn, &actual_size));
+            EXPECT_NOT_EQUAL(actual_size, 0);
+
+            /* Result matches actual size of data */
+            struct s2n_stuffer actual_data = { 0 };
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&actual_data, actual_size));
+            EXPECT_OK(s2n_tls13_serialize_resumption_state(conn, &actual_data));
+            EXPECT_EQUAL(actual_size, s2n_stuffer_data_available(&actual_data));
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+            EXPECT_SUCCESS(s2n_stuffer_free(&actual_data));
+        }
+    }
+
+    /* s2n_connection_get_session_length */
+    {
+        /* Safety */
+        EXPECT_EQUAL(s2n_connection_get_session_length(NULL), 0);
+
+        /* Session Ticket */
+        {
+            const uint16_t client_ticket_size = 10;
+
+            struct s2n_config *config = s2n_config_new();
+            EXPECT_NOT_NULL(config);
+            EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(config, true));
+
+            struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+            EXPECT_NOT_NULL(conn);
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+            EXPECT_SUCCESS(s2n_realloc(&conn->client_ticket, client_ticket_size));
+
+            uint8_t versions[] = { S2N_TLS12, S2N_TLS13 };
+            for (size_t i = 0; i < s2n_array_len(versions); i++) {
+                conn->actual_protocol_version = versions[i];
+                int session_length = s2n_connection_get_session_length(conn);
+                EXPECT_NOT_EQUAL(session_length, 0);
+
+                /* Result matches size expected by s2n_connection_get_session */
+                DEFER_CLEANUP(struct s2n_blob session_data = { 0 }, s2n_free);
+                EXPECT_SUCCESS(s2n_alloc(&session_data, session_length));
+                EXPECT_SUCCESS(s2n_connection_get_session(conn, session_data.data, session_data.size));
```

Comment:
Should the output of s2n_connection_get_session be tested here? Seems like we're just getting some data but not doing anything with it.

Reply:
Nope, not really anything to test against. We're really just interested in success: if session_data_size is too short, then the call will fail bc we won't be providing enough memory. I don't think there's any way to tell if session_data_size is too long, but I'm open to ideas xD